pkg/virt-handler/vm.go

```diff
@@ -605,6 +606,14 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 				// Remove the interface from domainInterfaceStatusByMac to mark it as handled
 				if interfaceStatus, exists := domainInterfaceStatusByMac[interfaceMAC]; exists {
 					newInterface.InterfaceName = interfaceStatus.InterfaceName
+
+					// Qemu guest reports CIDR, only an IP address should be passed to the status
+					err := iputils.ChangeCIDRsToIPs(&interfaceStatus)
```

Comment:
First of all that a lot for finding the issue and the contribution.

1. I find it confusing changing the cidrs to ips and not just returning the ips.
2. I would call the method only if `!isForwardingBindingInterface`.
3. The method should be invoked in the [un-handled](https://github.com/kubevirt/kubevirt/blob/b561ed9ede0eac44ef3667c75ece78c9538ed1b8/pkg/virt-handler/vm.go#L621) domain interfaces loop as well.

Anyway,
I think the correct place for the change is the [agent_parser](https://github.com/kubevirt/kubevirt/blob/aa8a15d540ad90d7022be59b30840d41581f52bc/pkg/virt-launcher/virtwrap/agent-poller/agent_parser.go#L268) and not here. IIUC, libvirt reports `ipAddr.IP` and `ipAddr.Prefix`. Our poller is covering it to 'ip\prefix`.
Since there may be users that are already using the cidr information I suggest that besides changing the poller `extractIPs` method, also adding the vmi api a new status field for the prefix.

Reply:
@EdDev what do you think about the suggestion here?

Reply:
I tend to agree that a better place to handle this is at the source, asking explicitly for the IP address (only) in the agent parser. The challenge here is how to keep both data available without duplicating code.

I am not raising here the issue about where and if we should expose the masks.